src/utilities/createRouter.ts

```diff
@@ -1,41 +1,70 @@
-import { readonly } from 'vue'
+import { DeepReadonly, reactive, readonly } from 'vue'
 import { Resolved, Route, RouteMethods, Routes } from '@/types'
 import { createRouteMethods, createRouterNavigation, resolveRoutes, routeMatch } from '@/utilities'
+import { isBrowser } from '@/utilities/isBrowser'
 import { resolveRoutesRegex } from '@/utilities/resolveRoutesRegex'
 
+type RouterOptions = {
+  initialUrl?: string,
+}
+
 type RouterPush = (url: string, options?: { replace: boolean }) => Promise<void>
 type RouterReplace = (url: string) => Promise<void>
 
 export type Router<
   TRoutes extends Routes
 > = {
   routes: RouteMethods<TRoutes>,
-  route: Readonly<Resolved<Route>>,
+  route: DeepReadonly<Resolved<Route>>,
   push: RouterPush,
   replace: RouterReplace,
   back: () => void,
   forward: () => void,
   go: (delta: number) => void,
 }
 
-export function createRouter<T extends Routes>(routes: T): Router<T> {
+export function createRouter<T extends Routes>(routes: T, options?: RouterOptions): Router<T> {
   const resolved = resolveRoutes(routes)
   const resolvedWithRegex = resolveRoutesRegex(resolved)
   const navigation = createRouterNavigation({
     onLocationUpdate,
   })
 
-  // todo: implement this
-  const route: Router<T>['route'] = readonly({} as any)
+  const route: Resolved<Route> = reactive(getInitialRoute())
```

Comment:
I think this makes sense, just wanted to call out that we could not support having an `undefined` router.route. So if something errors or we're handling not-found, we need to maybe assign this to some internal route that we defined as a fallback

Reply:
Correct. When a not found or a rejection happens we'll still need to create a resolved route. I've got some ideas for that and I might work on that soon.